butane_core/src/migrations/adb.rs

```diff
@@ -351,6 +351,32 @@ impl DeferredSqlType {
         }
     }
 }
+/// Compare, with Known and KnownId being identical if they contain the same type.
+impl PartialEq<DeferredSqlType> for DeferredSqlType {
+    fn eq(&self, other: &DeferredSqlType) -> bool {
+        match self {
+            Self::Known(sqltype) => {
+                if let Self::Known(other_sqltype) = other {
+                    return *sqltype == *other_sqltype;
+                }
+                // Replace self with KnownId, and run .eq again
+                other.eq(&Self::KnownId(sqltype.clone().into()))
+            }
+            Self::KnownId(type_id) => {
+                if let Self::KnownId(other_type_id) = other {
+                    return *type_id == *other_type_id;
+                }
+                false
```

Comment:
Don't we need to check if other is `Known`?

Reply:
I've added a manual `Eq` impl.
I thought there was a clippy lint that ensured that if `PartialEq` is manually implemented, `Eq` should also be manually implemented.  However I cant see it now in https://rust-lang.github.io/rust-clippy/master/index.html#/derive%20partialeq :/
Happy to switch back to deriving `Eq`.

Reply:
Deriving Eq seems slightly more succinct and produces the same result, but I don't feel strongly